File: clojure/problem1.clj
(ns rg.problem1)

(defn solve
	([] (solve 999 0))
	([n sum]
		(if (= n 1)
			sum
			(if (or (= 0 (mod n 3)) (= 0 (mod n 5)))
				(solve (- n 1) (+ sum n))
				(solve (- n 1) sum)))))

(println (solve))

File: erlang/problem1.erl
-module(problem1).
-export([solve/0]).
-import(lists).

solve() -> 
	lists:foldl(
		fun (X, Sum) -> X + Sum end, 0, 
		[ X || X <- lists:seq(1,999), (X rem 3 == 0) or (X rem 5 == 0) ]).
